dora/core/server/worker/src/main/java/alluxio/worker/netty/NettyReadHandlerStateMachine.java

```diff
@@ -672,6 +672,8 @@ private void readData(RequestContext requestContext, Transition<State, TriggerEv
             requestContext.getRequest());
       }
       fireNext(mTriggerEventsWithParam.mOutputLengthFulfilled, requestContext);
+      // Release the packet as there is nothing more to read
+      packet.release();
```

Comment:
i'm not really sure if it's best to release here or actually before the `fireNext()`. The `fireNext()` might throw an exception somehow? And I figure `fireNext()` is not using this packet anyways.

Or would there be a better place than here, to release the `packetReader.getDataBuffer()` instead of releasing on the temp variable like what I'm doing? Who cleans up the resources in the context?

Reply:
I think it is OK for this case, but I am concerning if there are other Bytebuf memory leak scenarios.